plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -0,0 +1,166 @@
+                        .setCatalogDirectory(getFileHiveMetastoreDirectory(queryRunner).toUri().toString())
+                        .setMetastoreUser("test"));
+
+        return queryRunner;
+    }
+
+    @Test
+    public void testV2TableRead()
+    {
+        String tableName = "test_v2_table_read" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation", 25);
+        updateTableToV2(tableName);
+        assertQuery("SELECT * FROM nation");
+    }
+
+    @Test
+    public void testV2TableWithRowBasedDelete()
+            throws Exception
+    {
+        String tableName = "test_v2_row_delete" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation", 25);
+        Table icebergTable = updateTableToV2(tableName);
+
+        String dataFilePath = (String) computeActual("SELECT file_path FROM \"" + tableName + "$files\" LIMIT 1").getOnlyValue();
+
+        Path metadataDir = new Path(getFileHiveMetastoreDirectory(getDistributedQueryRunner()).toUri());
+        String deleteFileName = "delete_file_" + UUID.randomUUID();
+        FileSystem fs = hdfsEnvironment.getFileSystem(new HdfsEnvironment.HdfsContext(SESSION), metadataDir);
+
+        PositionDeleteWriter<Record> writer = Parquet.writeDeletes(HadoopOutputFile.fromPath(new Path(metadataDir, deleteFileName), fs))
+                .forTable(icebergTable)
+                .overwrite()
+                .buildPositionWriter();
+
+        try (Closeable ignored = writer) {
+            writer.delete(dataFilePath, 0);
+        }
+
+        icebergTable.newRowDelta().addDeletes(writer.toDeleteFile()).commit();
+        assertQueryFails("SELECT * FROM " + tableName, "Iceberg tables with v2 deletes deltas are not supported");
+    }
+
+    @Test
+    public void testV2TableWithEqualityDelete()
+            throws Exception
+    {
+        String tableName = "test_v2_equality_delete" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation", 25);
+        Table icebergTable = updateTableToV2(tableName);
+
+        Path metadataDir = new Path(getFileHiveMetastoreDirectory(getDistributedQueryRunner()).toUri());
+        String deleteFileName = "delete_file_" + UUID.randomUUID();
+        FileSystem fs = hdfsEnvironment.getFileSystem(new HdfsEnvironment.HdfsContext(SESSION), metadataDir);
+
+        Schema deleteRowSchema = icebergTable.schema().select("regionkey");
+        EqualityDeleteWriter<Record> writer = Parquet.writeDeletes(HadoopOutputFile.fromPath(new Path(metadataDir, deleteFileName), fs))
+                .forTable(icebergTable)
+                .rowSchema(deleteRowSchema)
+                .createWriterFunc(GenericParquetWriter::buildWriter)
+                .equalityFieldIds(deleteRowSchema.columns().stream().filter(field -> field.name().equals("regionkey")).mapToInt(Types.NestedField::fieldId).toArray())
```

Comment:
why do we need to filter?

Reply:
It's not needed, strictly. I was just trying to mimic a `WHERE regionkey = x` delete instead of matching on the whole row. But since it just fails right now it doesn't really matter.

Reply:
doesn't `deleteRowSchema` already have only that column filtered? is there a reason to filter again?

Reply:
Ohh, yes sorry. I get what you're saying. Fixed